pathiomock.go

```diff
@@ -0,0 +1,59 @@
+package pathio
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"sync"
+)
+
+// MockClient mocks out an S3 bucket
+type MockClient struct {
```

Comment:
My intention was for this to be a more functional mock, but I guess we could go either way. If we do go the explicit mock route, I'd probably prefer to have it generated with [mock](https://github.com/golang/mock).